NeoOnnx/src/Optimization/LayerNormFusionOptimizer.cpp

```diff
@@ -0,0 +1,360 @@
+			&& transformLayer->GetDimensionRule( BD_BatchWidth ) == CTransformLayer::CDimensionRule( CTransformLayer::O_SetSize, 1 )
+			&& transformLayer->GetDimensionRule( BD_ListSize ) == CTransformLayer::CDimensionRule( CTransformLayer::O_InputDim, 0 )
+			&& transformLayer->GetDimensionRule( BD_Height ) == CTransformLayer::CDimensionRule( CTransformLayer::O_InputDim, 1 )
+			&& transformLayer->GetDimensionRule( BD_Width ) == CTransformLayer::CDimensionRule( CTransformLayer::O_SetSize, 1 )
+			&& transformLayer->GetDimensionRule( BD_Depth ) == CTransformLayer::CDimensionRule( CTransformLayer::O_SetSize, 1 )
+			&& transformLayer->GetDimensionRule( BD_Channels ) == CTransformLayer::CDimensionRule( CTransformLayer::O_InputDim, 3 ) )
+			||
+			( transformLayer->GetDimensionRule( BD_BatchLength ) == CTransformLayer::CDimensionRule( CTransformLayer::O_InputDim, 2 )
+			&& transformLayer->GetDimensionRule( BD_BatchWidth ) == CTransformLayer::CDimensionRule( CTransformLayer::O_InputDim, 3 )
+			&& transformLayer->GetDimensionRule( BD_ListSize ) == CTransformLayer::CDimensionRule( CTransformLayer::O_SetSize, 1 )
+			&& transformLayer->GetDimensionRule( BD_Height ) == CTransformLayer::CDimensionRule( CTransformLayer::O_InputDim, 6 )
+			&& transformLayer->GetDimensionRule( BD_Width ) == CTransformLayer::CDimensionRule( CTransformLayer::O_SetSize, 1 )
+			&& transformLayer->GetDimensionRule( BD_Depth ) == CTransformLayer::CDimensionRule( CTransformLayer::O_SetSize, 1 )
+			&& transformLayer->GetDimensionRule( BD_Channels ) == CTransformLayer::CDimensionRule( CTransformLayer::O_SetSize, 1 ) )
+			||
+			( transformLayer->GetDimensionRule( BD_BatchLength ) == CTransformLayer::CDimensionRule( CTransformLayer::O_InputDim, 0 )
+			&& transformLayer->GetDimensionRule( BD_BatchWidth ) == CTransformLayer::CDimensionRule( CTransformLayer::O_InputDim, 1 )
+			&& transformLayer->GetDimensionRule( BD_ListSize ) == CTransformLayer::CDimensionRule( CTransformLayer::O_SetSize, 1 )
+			&& transformLayer->GetDimensionRule( BD_Height ) == CTransformLayer::CDimensionRule( CTransformLayer::O_InputDim, 2 )
+			&& transformLayer->GetDimensionRule( BD_Width ) == CTransformLayer::CDimensionRule( CTransformLayer::O_SetSize, 1 )
+			&& transformLayer->GetDimensionRule( BD_Depth ) == CTransformLayer::CDimensionRule( CTransformLayer::O_SetSize, 1 )
+			&& transformLayer->GetDimensionRule( BD_Channels ) == CTransformLayer::CDimensionRule( CTransformLayer::O_SetSize, 1 ) );
+		return success ? transformLayer : nullptr;
+	}
+	return nullptr;
+}
+
+//--------------------------------------------------------------------------------------------------------------
+void CLayerNormFusionOptimizer::Apply()
+{
+	NeoAssert( graph.SelectionSize() == 0 );
+
+	int normFusionLayersCount = 0;
+	CArray<CBaseLayer*> layers{};
+	graph.GetLayers( layers );
+
+	for( int i = 0; i < layers.Size(); ++i ) {
```

Comment:
Now we can `for( CBaseLayer* layer : layers )`

Reply:
fixed